src/Diagnostics.DataProviders/DataProviders.cs

```diff
@@ -36,6 +38,7 @@ public DataProviders(DataProviderContext context)
                         throw new NotSupportedException($"{ds} is not supported.");
                 }
             };
+            MdmGeneric = (IMdmDataProviderConfiguration config) => new MdmLogDecorator(context, new MdmDataProvider(_cache, config, context.RequestId, context.receivedHeaders));
```

Comment:
Mdm = (MdmDataSource ds, IMdmDataProviderConfiguration? mdmConfiguration) =>
            {
				if(mdmConfiguration.Equals(null))
				{
					switch (ds)
					{
						case MdmDataSource.Antares:
							return new MdmLogDecorator(context, new MdmDataProvider(_cache, context.Configuration.AntaresMdmConfiguration, context.RequestId));
						case MdmDataSource.Networking:
							return new MdmLogDecorator(context, new MdmDataProvider(_cache, context.Configuration.NetworkingMdmConfiguration, context.RequestId));
						default:
							throw new NotSupportedException($"{ds} is not supported.");
					}
				}
				else
				{
					if(ds == MdmDataSource.Generic)
					{
						return new MdmLogDecorator(context, new MdmDataProvider(_cache, mdmConfiguration, context.RequestId, context.receivedHeaders));
					}
					else
					{
						throw new NotSupportedException($"{ds} is not supported.");
					}
				}
            };


The reason I am suggesting this is because if someone wants to use a Networking provider the detector code would look like
        dp.Mdm(MdmDataSource.Networking).someFunction 

Now if someone wants to use the generic Mdm, then the code would look like
               dp.MdmGeneric(mdmConfig).someFunction 

Instead, with the change I am suggesting, the code would be
For Networking : 
                  dp.Mdm(MdmDataSource.Networking).someFunction 
For any other Mdm provider 
                dp.Mdm(MdmDataSource.Generic, mdmConfig).someFunction 

There is consistency in the data provider usage and it is less confusing for people using the provider and the code is more readable.